backend/accounting-service/src/main/kotlin/services/grants/Gifts.kt

```diff
@@ -80,13 +80,15 @@ class GiftService(
 
             val resourceRequests = mutableListOf<GrantApplication.AllocationRequest>()
             var sourceProject = ""
+            var start = 0L
+            var end = 0L
             rows.forEach { row ->
                 val balance = row.getLong(0)!!
                 val category = ProductCategoryIdV2(row.getString(1)!!, row.getString(2)!!)
                 sourceProject = row.getString(3)!!
                 val renewalPolicy = row.getInt(4)!!
-                val start = DateTime.now().millis
-                val end = DateTime.now().plusMonths((if (renewalPolicy == 0) 12 else renewalPolicy)).millis
+                start = DateTime.now().millis
+                end = DateTime.now().plusMonths((if (renewalPolicy == 0) 12 else renewalPolicy)).millis
```

Comment:
Should this be overwritten for every iteration?

Reply:
There is no need for it to be overwritten because it would always be the same period, but since start and end is used further down for submit revision and renewalPolicy is only know when read from the row I just did this. 

This is most likely also legacy code since either the GrantApplication.Period is not used from the revision submitted or the Period from the resourceRequest is not used when creating the resources.

Alternative code: I can read the first entry in the rows to get the renewal policy to calculate the start and end before iterating all rows for resouceRequests